helion/_compiler/generate_ast.py

```diff
@@ -74,6 +74,18 @@ def add_statement(self, stmt: ast.AST | str | None) -> None:
             stmt = statement_from_string(stmt)
         self.statements_stack[-1].append(stmt)
 
+    def get_rng_seed_buffer_statements(self) -> list[ast.AST]:
+        import_stmt = statement_from_string(
+            "from torch._inductor import inductor_prims"
+        )
+
+        # Create host-side seed buffer with the required number of seeds
+        seed_buffer_stmt = statement_from_string(
+            f"_rng_seed_buffer_host = inductor_prims.seeds({self.device_function.rng_seed_count}, torch.device('cuda'))"
```

Comment:
Here we use the same approach as torch.compile (allocate one seed value per RNG op).